<a href="https://colab.research.google.com/github/YusolCho/Game_Analysis/blob/main/LoL_Match_majority_practice.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# gold tier user들의 summoner name과 summoner id를 데이터 프레임 형식으로 저장


In [ ]:
import requests
import pandas as pd

# Replace {api_key} with your Riot API key
api_key = 'RGAPI-5f1c9482-d894-4470-b2a1-ebd2934b10bf'

# Set up the base URL for the Riot API
base_url = 'https://{region}.api.riotgames.com'

# Set up the regions you're interested in
regions = ['kr']

# Set up the rank and division criteria
rank = 'GOLD'
division = 'I'

# Set up an empty list to store the user data
user_data = []

# Iterate over each region
for region in regions:
    # Set up the League API endpoint URL
    league_url = base_url.format(region=region) + '/lol/league-exp/v4/entries/RANKED_SOLO_5x5/{tier}/{division}'
    # Set up the headers for the API requests
    headers = {'X-Riot-Token': api_key}
    
    # Make a request to the League API for users who meet the criteria
    response = requests.get(league_url.format(tier=rank, division=division), headers=headers)
    league_data = response.json()
    
    # Iterate over each user and collect their data
    for user in league_data:
        user_dict = {
            'summoner_id' : user['summonerId'],
            'summoner_name': user['summonerName'],
#            'wins': user['wins'],
#           'losses': user['losses'],
#            'win_ratio': user['wins'] / (user['wins'] + user['losses'])
        }
        user_data.append(user_dict)

# Convert the list of user data to a pandas DataFrame
df = pd.DataFrame(user_data)

# Print the DataFrame
print(df)

                                          summoner_id  summoner_name
0    xn87B73DmRFH_2m42rX7B9eTe9kWWMKMSG97yXO2UlMf2lep         J꾸짖을 갈
1    QPi-8CUDhm_mYttzQKpDR3EYhgRraB05V-DpX21HVmwN3DYG            푸 투
2    wO6e2sgb8s5qexhWcAOV60yTn6cppq4kVTNLnB4P9HqhKqGP       마법소녀 모야삡
3    avNBhDDrbGeoKshj0k4RBSgKU7zXgJA0p60UbXvDFgNLtvkh           응애소나
4    YRpk_hdXNUS_7QNh3h9VV5ReoiMcxvLSMK1_PTsWYdNyDarA          막타의호흡
..                                                ...            ...
200  HzvETRuNiP3qQgK0MItYrMlfE9fyA1_SlqaoU0opTIj7qcI3        JHBRANT
201  XmxRPW5_nhwSHdRER-hb9ckMDYhN7wXUim9D-wPUe1py7Txn   MEVIUS CASSI
202  p5IUvbhvXESUosUVZREmo5EwnkjVDGJLnFfY42nfX8TvDCHn          이잉민린이
203   vmaHPA8ekd_Ot37FOtA-zi47XSjemLw-WolYm2Ac_h2qYfY         홍콩헹게이바
204  ccco7nPoBq3dQ2vj-xLTBSADv7KOgnLTzG9FN4iMSYd_HdYK        나오늘막나갈래

[205 rows x 2 columns]


# 위 데이터 프레임에서 summoner id만 리스트로 만들기

In [ ]:
# Extract the 'summoner_id' column from the DataFrame as a pandas Series
summoner_id_series = df['summoner_id']

# Convert the Series to a list
summoner_id_list = summoner_id_series.tolist()

# Print the resulting list
print(summoner_id_list)


['xn87B73DmRFH_2m42rX7B9eTe9kWWMKMSG97yXO2UlMf2lep', 'QPi-8CUDhm_mYttzQKpDR3EYhgRraB05V-DpX21HVmwN3DYG', 'wO6e2sgb8s5qexhWcAOV60yTn6cppq4kVTNLnB4P9HqhKqGP', 'avNBhDDrbGeoKshj0k4RBSgKU7zXgJA0p60UbXvDFgNLtvkh', 'YRpk_hdXNUS_7QNh3h9VV5ReoiMcxvLSMK1_PTsWYdNyDarA', 'ivGNQHWrww9HGNVvidlMR6nHYAYPXsduMXZenyc9Yeho4hSS', 'BNCP-z1GdtORGDJHgGtDqeoaVLjSDnkn12T1fXziPZsNuIU8', 'RXPcjiUmBFwCTEMZRe4acW2SFevtgTV386otWA_ACfYcfQEo', 'eoliDpTZ4Vim88RkuXBeq-eHuEyl7Ideohj_0k7CF7niKZ4', 'uJrIczRq5mkBWECBe-_54aslKZPRKJcFATYdr2RkGQydTRw', 'dD0xBiYBvTEf72R0jeOrPGS7xvFiJxP7JNrWpc1K1PjswLo', 'a4mStmh-oFCkAZGURWsXcZFWs3j-PfG_p3nSlGAMYiDavBM', 'pa-iqoYBBBwdzgbqJoFC9a9_Re7PMVF2gXHNETnPW9c9YCs', 'zbtl5B4PsMtrgyWTv6BliH94gIBqXo4TOaq0pmG4MkmLi8gw', 'kOim0v43FmmF6BPgy2rl6yhjSkl8y32waan5yePcAq8Ddq3C', 'XYw9H4STjyVDaq3ucgIPc_3uySM4mLPgs3CJxbSWE8N-mhGU', '5APMU6HrYvZtP-DfGxMbXEukjDaFjTCda6PIRFFq_EvikeNn', 'lRKSu-pU4AJx_C8DVa87A2QiaPfb7uiQV1PpR127NS81i7E', 'np8voCa6tIo0lgvvOUMoUOoFZDx9f6yIYd_6WnmYosdb9Q9H', 'RAs9O08X6pbdwXkv

# 추출된 summoner id 리스트에 해당되는 puuid 리스트 만들기

In [ ]:
# 4분 걸림 ... 

import requests
import time
import retrying

# Replace {api_key} with your Riot API key
api_key = 'RGAPI-5f1c9482-d894-4470-b2a1-ebd2934b10bf'

# Set up the base URL for the Riot API
base_url = 'https://kr.api.riotgames.com'

# Set up the regions you're interested in
regions = ['kr']

# Set up an empty list to store the puuids
puuid_list = []

# Set up rate limit variables
RATE_LIMIT_RETRIES = 5
RATE_LIMIT_BACKOFF_FACTOR = 2

# Set up retries variables
RETRIES = 5
BACKOFF_FACTOR = 2

# Define a function to make a request to the Riot API with retries and rate limiting
@retrying.retry(wait_exponential_multiplier=1000, wait_exponential_max=10000, stop_max_attempt_number=RETRIES)
def request_riot_api(url, headers):
    response = requests.get(url, headers=headers)
    if response.status_code == 429:
        retry_after = int(response.headers['Retry-After'])
        time.sleep(retry_after + 1)
        raise Exception('Rate limit exceeded, retrying...')
    elif response.status_code != 200:
        raise Exception(f'Request failed with status code {response.status_code}: {response.content.decode()}')
    return response.json()

# Iterate over each region
for region in regions:
    # Set up the Summoner API endpoint URL
    summoner_url = base_url.format(region=region) + '/lol/summoner/v4/summoners/{encryptedSummonerId}'
    
    # Set up the headers for the API requests
    headers = {'X-Riot-Token': api_key}
    
    # Iterate over each summoner ID in the list
    for summoner_id in summoner_id_list:
        # Make a request to the Summoner API for the corresponding puuid
        for i in range(RATE_LIMIT_RETRIES):
            try:
                summoner_data = request_riot_api(summoner_url.format(encryptedSummonerId=summoner_id), headers)
                break
            except Exception as e:
                print(e)
                print(f'Retrying in {BACKOFF_FACTOR ** i} seconds...')
                time.sleep(BACKOFF_FACTOR ** i)

        # Append the puuid to the list
        puuid_list.append(summoner_data['puuid'])

# Print the resulting puuid list
print(puuid_list)

['YvRwtybq9iG4ybIeLn9osnfzrUPob5bNgcVkOwWJd5jm9hs70w0jy9_mdH3VCwekIzM1vnrk7rq8Kg', 'ev9cR3_z9HIDe-BbdMP8KcSQb1micGugw_ogBcCWUdUI4cyrrV6Np3ozKtFexJIkJ0gc47CCsl6AUg', '7dGrIHqteZzlicQ9pgm80swBRNzmV16qYFeVnmRZBYrkotQxCzBme84SDGVjMQP5xn-qAn8Vdqx4Hw', 'KcjxziNxPL21KTZTnW0Uyp4Mg-uW36xuJZCA43qte_AFG_nQ6Ti8V_MBH8laS9-dLF8RKJimW-Yoew', 'uQlxMetzKy8xUqKvrTtNmSHU4zBKGihpZA20fcwd7Ao1lcGCKJBhU2QyRhkm9i-V5OwiNNOSaRoCJQ', 'AXLhMB-GcfRoFrmFSHSosnGXZhvAYAgW55FeZX_8v6QCGlM9UC1snKNcAr_z-ag38PBnPY_hAqUKzQ', 'dlSZLefbV8gYfAlDw87WO3a1de1OZpQ4ieGcyHHOKyBTWdX7bBPwqo-SIsAmE6XxCh-39kGAPk0K-Q', '89KMebQJVJ9h2NpQjQsdQU_CDlOR3wgYHv0rcCcfM8T7cULGCvE2_gj9QpjHpbIcHa6BC0YxE7tMZA', 'Ci1TFrPtdo8xEL5tLaVCmZAIzv7cbjf-aRnIJg8pWGpv7C1B2p258sgQF517sNT0xr38uADw2wboWQ', '1BqmT4PHg-HI_RNCV1w9cY6GRGK2Li-Dj38hsH2VGuKk8zmDJ2toVQTgJqpciYaLzP2pEtDdVFsf9w', 's8XrlAjztzTANMZSj1H7Qy6DEbD6-1gcTyQFjlKpTPEI_Eaqkw2b0HNSyY-s314NiznX7xuAsEY96g', 'm5ltXnzvqu3b7LLxNZuqr_K8jlKsaSX8nVFJ2igsUMg_J1H__h_ajlqrDahDuOPeG5hiL77Yd_De-g', 'kQD_WQJIMpovJr

In [ ]:
#!pip install retrying

Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/


# puuid 리스트를 csv파일로 저장하기 

In [ ]:
import csv
f = open('puuidfile.csv', 'w')
writer = csv.writer(f)
writer.writerow(puuid_list)
f.close()

In [ ]:
with open('puuidfile.csv','r',newline='') as f:
    reader = csv.reader(f)
    for row in reader:
        print(row)

['YvRwtybq9iG4ybIeLn9osnfzrUPob5bNgcVkOwWJd5jm9hs70w0jy9_mdH3VCwekIzM1vnrk7rq8Kg', 'ev9cR3_z9HIDe-BbdMP8KcSQb1micGugw_ogBcCWUdUI4cyrrV6Np3ozKtFexJIkJ0gc47CCsl6AUg', '7dGrIHqteZzlicQ9pgm80swBRNzmV16qYFeVnmRZBYrkotQxCzBme84SDGVjMQP5xn-qAn8Vdqx4Hw', 'KcjxziNxPL21KTZTnW0Uyp4Mg-uW36xuJZCA43qte_AFG_nQ6Ti8V_MBH8laS9-dLF8RKJimW-Yoew', 'uQlxMetzKy8xUqKvrTtNmSHU4zBKGihpZA20fcwd7Ao1lcGCKJBhU2QyRhkm9i-V5OwiNNOSaRoCJQ', 'AXLhMB-GcfRoFrmFSHSosnGXZhvAYAgW55FeZX_8v6QCGlM9UC1snKNcAr_z-ag38PBnPY_hAqUKzQ', 'dlSZLefbV8gYfAlDw87WO3a1de1OZpQ4ieGcyHHOKyBTWdX7bBPwqo-SIsAmE6XxCh-39kGAPk0K-Q', '89KMebQJVJ9h2NpQjQsdQU_CDlOR3wgYHv0rcCcfM8T7cULGCvE2_gj9QpjHpbIcHa6BC0YxE7tMZA', 'Ci1TFrPtdo8xEL5tLaVCmZAIzv7cbjf-aRnIJg8pWGpv7C1B2p258sgQF517sNT0xr38uADw2wboWQ', '1BqmT4PHg-HI_RNCV1w9cY6GRGK2Li-Dj38hsH2VGuKk8zmDJ2toVQTgJqpciYaLzP2pEtDdVFsf9w', 's8XrlAjztzTANMZSj1H7Qy6DEbD6-1gcTyQFjlKpTPEI_Eaqkw2b0HNSyY-s314NiznX7xuAsEY96g', 'm5ltXnzvqu3b7LLxNZuqr_K8jlKsaSX8nVFJ2igsUMg_J1H__h_ajlqrDahDuOPeG5hiL77Yd_De-g', 'kQD_WQJIMpovJr

# 리스트 내 puuid에 해당되는 유저의 매치id 불러오기. 각 유저당 최근 1판만.
* 총 205개의 매치 id가 추출됨
* 데이터 개수가 적다. 나중에 각 유저당 10개, 50개 ... 로 늘리면 데이터가 많아질 것.
* 현재 총 205개의 puuid 가 있음
* MID에에 해당되는 것만 추출하면 데이터 양이 줄 것
* 


In [ ]:
import requests
import time
import json

# Replace {api_key} with your Riot API key
api_key = 'RGAPI-5f1c9482-d894-4470-b2a1-ebd2934b10bf'

# Set up the base URL for the Riot API
base_url = 'https://asia.api.riotgames.com'

# Set up the regions you're interested in
regions = ['asia']

# Set up an empty dictionary to store the match ids
match_id_dict = {}

# Set up a cache to store API responses
response_cache = {}

# Define a function to make a request to the Riot API
def request_riot_api(url, headers):
    # Check if the response is in the cache
    if url in response_cache:
        return response_cache[url]
    
    # Make the API request
    response = requests.get(url, headers=headers)
    
    # Check if the rate limit has been exceeded
    if response.status_code == 429:
        # If the rate limit has been exceeded, wait for the retry-after time and try again
        retry_after = response.headers.get('Retry-After')
        time.sleep(int(retry_after))
        response = requests.get(url, headers=headers)
    
    # Raise an exception if the request fails
    if response.status_code != 200:
        raise Exception(f'Request failed with status code {response.status_code}: {response.content.decode()}')
    
    # Cache the response
    response_cache[url] = response.json()
    
    return response.json()

# Iterate over each region
for region in regions:
    # Set up the Match API endpoint URL
    match_url = base_url.format(region=region) + '/lol/match/v5/matches/by-puuid/{puuid}/ids?start=0&count=1'
    
    # Set up the headers for the API requests
    headers = {'X-Riot-Token': api_key}
    
    # Iterate over each puuid in the list
    for puuid in puuid_list:
        # Make a request to the Match API for the latest match id for the user
        match_data = request_riot_api(match_url.format(puuid=puuid), headers)
        
        # Add the latest match id to the dictionary
        if match_data:
            match_id_dict[puuid] = match_data[0]

# Save the response cache to a file
with open('response_cache.json', 'w') as f:
    json.dump(response_cache, f)

# Print the resulting dictionary of match ids
print(match_id_dict)

{'YvRwtybq9iG4ybIeLn9osnfzrUPob5bNgcVkOwWJd5jm9hs70w0jy9_mdH3VCwekIzM1vnrk7rq8Kg': 'KR_6414892833', 'ev9cR3_z9HIDe-BbdMP8KcSQb1micGugw_ogBcCWUdUI4cyrrV6Np3ozKtFexJIkJ0gc47CCsl6AUg': 'KR_6415496326', '7dGrIHqteZzlicQ9pgm80swBRNzmV16qYFeVnmRZBYrkotQxCzBme84SDGVjMQP5xn-qAn8Vdqx4Hw': 'KR_6412274821', 'KcjxziNxPL21KTZTnW0Uyp4Mg-uW36xuJZCA43qte_AFG_nQ6Ti8V_MBH8laS9-dLF8RKJimW-Yoew': 'KR_6415542485', 'uQlxMetzKy8xUqKvrTtNmSHU4zBKGihpZA20fcwd7Ao1lcGCKJBhU2QyRhkm9i-V5OwiNNOSaRoCJQ': 'KR_6412399348', 'AXLhMB-GcfRoFrmFSHSosnGXZhvAYAgW55FeZX_8v6QCGlM9UC1snKNcAr_z-ag38PBnPY_hAqUKzQ': 'KR_6403121875', 'dlSZLefbV8gYfAlDw87WO3a1de1OZpQ4ieGcyHHOKyBTWdX7bBPwqo-SIsAmE6XxCh-39kGAPk0K-Q': 'KR_6406038094', '89KMebQJVJ9h2NpQjQsdQU_CDlOR3wgYHv0rcCcfM8T7cULGCvE2_gj9QpjHpbIcHa6BC0YxE7tMZA': 'KR_6415369842', 'Ci1TFrPtdo8xEL5tLaVCmZAIzv7cbjf-aRnIJg8pWGpv7C1B2p258sgQF517sNT0xr38uADw2wboWQ': 'KR_6411430665', '1BqmT4PHg-HI_RNCV1w9cY6GRGK2Li-Dj38hsH2VGuKk8zmDJ2toVQTgJqpciYaLzP2pEtDdVFsf9w': 'KR_6415473192', 's8XrlAjz

In [ ]:
# create a DataFrame with puuid as index and match_id as column
match_id_df = pd.DataFrame({"match_id": match_id_dict})
match_id_df.index.name = "puuid"
     

In [ ]:
print(match_id_df)  #추출 완료

                                                         match_id
puuid                                                            
-48wVkHbkDlbEGy-lqd10qebfRLH_UIw-KTn_ZdG_g82pSk...  KR_6415103707
-D3u-xcIv4VaKAwWzMZXAxGEvhnJHqlUL9AuJCJv_vKQK-_...  KR_6414716198
-_X_4zdqcB_7fwOOU7tcfduWZX0Q7EZ5qttXtOhZlkXIzfb...  KR_6413205595
-omeeLCMgQo-1o4SyjTLh3BuBPiVfdbMipdYTA8cKX4c7y5...  KR_6344934892
-sqokPlQPjwtG3KSjEuTHaSBFQugk8VoBM6ZmIGJur4iScw...  KR_6414141120
...                                                           ...
y5OooQBuBPdQkhjYPvrICrFB_bf46LDHc6BEZdzyadfUK5D...  KR_6415489595
ybz0BUwivog3Q-jFGsgaonw40oGW3naI8-ax19Yyc6LmpJN...  KR_6403356409
yooz0DNy9L3_Q-qcBGSoE9vWpS4vjwTx-da1s1DO_-5IHdx...  KR_6384104090
z681-WywGM5MnZSoKqyXvASsUq2rLLV0DThB5q0T2hvmkfj...  KR_6415645130
ziTPMDvFa0Bf3KvFNuU3KP_xJQ39vmPUcQExkEpqzQPkuUo...  KR_6349519426

[205 rows x 1 columns]


In [ ]:
# Extract the 'summoner_id' column from the DataFrame as a pandas Series
match_id_series = match_id_df['match_id']

# Convert the Series to a list
match_id_list_ = match_id_series.tolist()

# Print the resulting list
print(match_id_list_)

['KR_6415103707', 'KR_6414716198', 'KR_6413205595', 'KR_6344934892', 'KR_6414141120', 'KR_6342406573', 'KR_6386684535', 'KR_6411249673', 'KR_6415473192', 'KR_6415640878', 'KR_6408239878', 'KR_6393387652', 'KR_6415767934', 'KR_6411593939', 'KR_6404315710', 'KR_6393706255', 'KR_6415331310', 'KR_6414629973', 'KR_6415495880', 'KR_6410170112', 'KR_6385361795', 'KR_6328618574', 'KR_6415776141', 'KR_6415344383', 'KR_6330457917', 'KR_6415479452', 'KR_6412102293', 'KR_6392455021', 'KR_6414606879', 'KR_6415198765', 'KR_6388135961', 'KR_6415284011', 'KR_6402687539', 'KR_6397657869', 'KR_6412274821', 'KR_6415369842', 'KR_6411207733', 'KR_6390594788', 'KR_6415364955', 'KR_6415540840', 'KR_6414145213', 'KR_6411235767', 'KR_6415508445', 'KR_6415051557', 'KR_6403121875', 'KR_6413777984', 'KR_6414874829', 'KR_6404281772', 'KR_6413715254', 'KR_6399872118', 'KR_6334922659', 'KR_6415742857', 'KR_6411430665', 'KR_6412949135', 'KR_6414169497', 'KR_6415460489', 'KR_6408779667', 'KR_6412543525', 'KR_641273181

# 각 매치 id별로 매치 meta data 불러오기

In [ ]:
# 예슬이 코드 
# Replace {api_key} with your Riot API key
api_key = 'RGAPI-5f1c9482-d894-4470-b2a1-ebd2934b10bf'

#<info>에 있는 변수
gameMode = []
gameType = []

#<participants>에 있는 변수
assists = []
kills = []
deaths = []
champExp = []
champName = []
indiPos = []
teamPos = []
lane = []
role = []
gold = []
tDDealt = []
tDDealtChamp = []
tDTaken = []
tMKilled = []
turKills = []
win = []

for i in match_id_list:
    url = "https://asia.api.riotgames.com/lol/match/v5/matches/" + i + "?api_key=" + api_key
    r = requests.get(url)
    r = r.json()
    info = r["info"] #전체 데이터에서 info를 추출
    part = info["participants"] #info 데이터에서 참가자들의 정보 추출
    gameMode.append(info["gameMode"])
    gameType.append(info["gameType"])
    
    for j in range(10): #총 10명의 유저 중 특정 puuid를 가진 유저를 추출
        if part[j]["puuid"] == puu_id:
            assists.append(part[j]["assists"])
            kills.append(part[j]["kills"])
            deaths.append(part[j]["deaths"])
            champExp.append(part[j]["champExperience"])
            champName.append(part[j]["championName"])
            indiPos.append(part[j]["individualPosition"])
            teamPos.append(part[j]["teamPosition"])
            lane.append(part[j]["lane"])
            role.append(part[j]["role"])
            gold.append(part[j]["goldEarned"])
            tDDealt.append(part[j]["totalDamageDealt"])
            tDDealtChamp.append(part[j]["totalDamageDealtToChampions"])
            tDTaken.append(part[j]["totalDamageTaken"])
            tMKilled.append(part[j]["totalMinionsKilled"])
            turKills.append(part[j]["turretKills"])
            win.append(part[j]["win"])

In [ ]:
match_id_list_mini = match_id_list_[1:5]
print(match_id_list_mini)

['KR_6414716198', 'KR_6413205595', 'KR_6344934892', 'KR_6414141120']


In [ ]:
#챗지피티+예슬
import requests
import pandas as pd

api_key = 'RGAPI-5f1c9482-d894-4470-b2a1-ebd2934b10bf'
region = 'asia'

# Empty lists to store the collected data
game_mode_list = []
game_type_list = []
assists_list = []
kills_list = []
deaths_list = []
champ_experience_list = []
champion_name_list = []
individual_position_list = []
lane_list = []
role_list = []
gold_earned_list = []
team_position_list = []
total_damage_dealt_list = []
total_damage_dealt_to_champions_list = []
total_damage_taken_list = []
total_minions_killed_list = []
turret_kills_list = []
win_list = []

# Loop through each match ID and collect data using the Riot API
for match_id in match_id_list_mini:
    url = f'https://asia.api.riotgames.com/lol/match/v5/matches/{match_id}?api_key={api_key}'
    r = requests.get(url)
    r = r.json()
    info = r["info"] #전체 데이터에서 info를 추출
    gameMode.append(info["gameMode"])
    gameType.append(info["gameType"])
    
    for j in range(10): #총 10명의 유저 중 특정 puuid를 가진 유저를 추출
        if part[j]["puuid"] == puu_id:
            assists.append(part[j]["assists"])
            kills.append(part[j]["kills"])
            deaths.append(part[j]["deaths"])
            champExp.append(part[j]["champExperience"])
            champName.append(part[j]["championName"])
            indiPos.append(part[j]["individualPosition"])
            teamPos.append(part[j]["teamPosition"])
            lane.append(part[j]["lane"])
            role.append(part[j]["role"])
            gold.append(part[j]["goldEarned"])
            tDDealt.append(part[j]["totalDamageDealt"])
            tDDealtChamp.append(part[j]["totalDamageDealtToChampions"])
            tDTaken.append(part[j]["totalDamageTaken"])
            tMKilled.append(part[j]["totalMinionsKilled"])
            turKills.append(part[j]["turretKills"])
            win.append(part[j]["win"])

SyntaxError: ignored

# 하나의 매치 데이터 출력해보기 

In [ ]:
single_match_id = 'KR_6414892833'
api_key = 'RGAPI-5f1c9482-d894-4470-b2a1-ebd2934b10bf'
region = 'asia'
match_id = single_match_id
url = f'https://asia.api.riotgames.com/lol/match/v5/matches/{match_id}?api_key={api_key}'
r = requests.get(url)
r = r.json()

In [ ]:
type(r)

dict

In [ ]:
r

{'metadata': {'dataVersion': '2',
  'matchId': 'KR_6414892833',
  'participants': ['NRDTJMBAHKzfX5dJlFzdrFnG8zr4QvrCSqD5c7r3sdvyvLL6wc2wMu7Zy8cOVBcLwApy2kixwuxL-g',
   'avLaN-73Vc60gN9Jt9EKzx2tfuCWi7DOy_25q4E40PACJ9HoqO4wB2ai6jmLNcHNlKHajVurwjuvSA',
   'SQLSZqbdxYieqP2BdGVC4zLSXo8vndSqZJS1Lpa_OSUvV2OlO1cIjig7hndpk-69czvlifhVucgeow',
   'lT5Hd08sd1ozu3gzWgNPHmzmLApUKHu9sr_baVZWPdIioXFLDRSFVKi2vboeU0OEOp2Uq3wMDSnlgw',
   'A0xAwVr6He3E8cZcl5AyNto26c7BDN8O-xplUEKglGahk-w9TiEST6NR7-vvLyoMvCta_RV2rGBRHw',
   'DVL5VMoyjilnF2m_fh2mPKsfgLUGdjwolqIvPV_bK0bXXXW1CDC2EreINOQiM0vBeCle-8UXzGk2cw',
   'nnD6jsAcC26gJdXXi5_wCPHSB7wzPuv9gKPUBsz2aq6bmHjNQHjJv1GGOUTtwQIBJ2KRhVaChxkF3Q',
   'm8rAf95YOKx8is_eF7bdp5vRxiZsFMDqc4CfkTSnmNGjtORJbdRl0XPTh7pdP6XIga6WRmq-9mD7aQ',
   'YvRwtybq9iG4ybIeLn9osnfzrUPob5bNgcVkOwWJd5jm9hs70w0jy9_mdH3VCwekIzM1vnrk7rq8Kg',
   'CGMVmtEcRjVU66OS9VRFUItcHWzyvur1F-XQHL6Y-tjox6DaPK1Umq8Yhd-d2PWN7QnVskAhfR8EZw']},
 'info': {'gameCreation': 1679393731936,
  'gameDuration': 1537,
  '

In [ ]:
r['metadata']['matchId']

'KR_6413205595'

In [ ]:
r['info']['gameMode']
r['info']['gameType']

In [ ]:
r['info']['participants']

In [ ]:
len(r['info']['participants'])

10

#### 어려운 점이 생김 
* match-v5 에다가 하나의 match id 를 넣었을 때 나오는 정보는 크게 meta data, info data, participants data 인데, 문제는 이 participants 데이터에는 이 게임에 참가한 모든 플레이어들의 데이터가 들어있어서 우리가 원하는 특정 puuid의 유저를 뽑으려면 코드가 좀 복잡해질 것 같음. 
* 일단, '통합'이 가장 중요할 것 같음. summoner name, summoner id, puuid, match id 를 모두 하나의 데이터 프레임, 혹은 딕셔너리로 저장해야한다. -> 가시성이 좋은 데이터프레임으로 바꾸자. 


# puuid, match id 를 하나의 데이터 프레임으로 통합하기 
* 지금은 각 puuid당 하나의 match id 를 뽑아서 그냥 가로로 붙여주면 되지만, 나중에 한 puuid당 여러개의 match id(50개, 100개, ... )를 뽑는다면 그냥 막 붙이면 안됨 

#### **!!!! 주의 !!!!**
#### 아래의 데이터프레임 쓰지 말것. 아무래도 puuid와 match id가 서로 매칭되지 않는 것 같음. 대신, 위에 만들어놓았던 match_id_dict(딕셔너리 꼴) 사용하기 

In [ ]:
#puu_matchid_df = pd.DataFrame()
#puu_matchid_df['puuid'] = puuid_list
#puu_matchid_df['match_id'] = match_id_list_
#puu_matchid_df.head(10)
#mini dataframe
#puu_matchid_df_mini = puu_matchid_df.head(10)


puuid       match_id
0  YvRwtybq9iG4ybIeLn9osnfzrUPob5bNgcVkOwWJd5jm9h...  KR_6415103707
1  ev9cR3_z9HIDe-BbdMP8KcSQb1micGugw_ogBcCWUdUI4c...  KR_6414716198
2  7dGrIHqteZzlicQ9pgm80swBRNzmV16qYFeVnmRZBYrkot...  KR_6413205595
3  KcjxziNxPL21KTZTnW0Uyp4Mg-uW36xuJZCA43qte_AFG_...  KR_6344934892
4  uQlxMetzKy8xUqKvrTtNmSHU4zBKGihpZA20fcwd7Ao1lc...  KR_6414141120
5  AXLhMB-GcfRoFrmFSHSosnGXZhvAYAgW55FeZX_8v6QCGl...  KR_6342406573
6  dlSZLefbV8gYfAlDw87WO3a1de1OZpQ4ieGcyHHOKyBTWd...  KR_6386684535
7  89KMebQJVJ9h2NpQjQsdQU_CDlOR3wgYHv0rcCcfM8T7cU...  KR_6411249673
8  Ci1TFrPtdo8xEL5tLaVCmZAIzv7cbjf-aRnIJg8pWGpv7C...  KR_6415473192
9  1BqmT4PHg-HI_RNCV1w9cY6GRGK2Li-Dj38hsH2VGuKk8z...  KR_6415640878

In [ ]:
# 딕셔너리로 수정 완료! 
# 미니 만들어서 실험해보면 됨 
# response.content 해봤는데... 결과가 없다고 뜸.
# 아마도 API 기간 만료된 듯.
api_key = 'RGAPI-5f1c9482-d894-4470-b2a1-ebd2934b10bf'
region = 'asia'
assists_list = []

for i in range(0,30):
    pair = list(match_id_dict.items())[i]
    puuid = pair[0]
    match_id = pair[1]
    url = f'https://asia.api.riotgames.com/lol/match/v5/matches/{match_id}?api_key={api_key}'

    # Make the Riot API call
    response = requests.get(url)
    response_json = response.json()

    assists = None  # set a default value for assists
    for participant in response_json['info']['participants']:
        if participant['puuid'] == puuid:
            assists = participant['assists']
            break

    if assists is not None:
        assists_list.append(assists)  # append assists only if it was set to a non-default value


In [ ]:
response.content

b'{"metadata":{"dataVersion":"2","matchId":"KR_6375298403","participants":["fUg259_KyFQ8DYzSvUBVPiu4xeD6Ia-b4dePpV0xP8pK-9wUfWj1Lp5ZhVcgY-Djxic1nLg8K0MEhw","9sQhKMYf-BS0uyrgTW3KNr22xNb3SjcP4oQCQELbPZU6TUFj6H6DWbCPHyaRPBWCRqvyAZKtCOOQWQ","AbaN8DFHXf0e-YXk6av2fZhH-3ndLc4OAxnJjHyLBe8x82i37RPELuFoAIQCgN08TxxcmQfjrq9IFg","boT4_kfK0OK3FIzo8x345iva-QoUb1SMpHa_bXU62ohTKZe_KZBpia7Lb_DcQ0-_EWrTF1V0ZXkTzw","d35prQAlgWC9e9tmQ7H5S-IgetuAZSngmKQdJ039DLIsDuAJPJefEKrOXOkCeWV_I-Ks3IpQ2GrYbQ","wMTHyk6cTf189-aZhgLLZY5R6FjgRybLQPCiDtiV4RQLwQ8hkY8LoSCiOOlHhGVSJgq-S41qRle4qQ","_k4lEbV2aogRPqSflM9b_Cazo-kJVWXD9vf8LprdTnGPPp_de2r3F3J67Q2gG-K-EvLceyLDYNsN2g","zVTVLP-z8tQDabW-9Me6UXgCX578rcj4FnND8gMvc-3aE7YIFNL5xWP6xRa1_bF2178cBEvKTvHGgw","v1QT12Pn-vU8eJxQbS4kftGig7OVs5lz3iSEbG3U_x50YwHXkm1Qnw30jublEVOHG2gXH_fpo2hHoA","NQRRMoP514CFqE39Zk7r8OeOMaLAnxQzyf7W-SDzvOoDHYTBHtyz222q-0pdTb8uyMcMdCGEOGA2iA"]},"info":{"gameCreation":1677061118121,"gameDuration":2024,"gameEndTimestamp":1677063183533,"gameId":6375298403,"ga

In [ ]:
assists_list  #성공!

[3,
 1,
 2,
 2,
 15,
 7,
 7,
 3,
 19,
 15,
 9,
 22,
 3,
 2,
 5,
 10,
 5,
 25,
 1,
 6,
 4,
 3,
 34,
 33,
 28,
 10,
 7,
 2,
 16,
 12]

In [ ]:
# 딕셔너리로 수정 완료! 
# 미니 만들어서 실험해보면 됨 
# response.content 해봤는데... 결과가 없다고 뜸.
# 아마도 API 기간 만료된 듯.
api_key = 'RGAPI-5f1c9482-d894-4470-b2a1-ebd2934b10bf'
region = 'asia'
# Empty lists to store the collected data
# game_mode_list = [] 얘넨 나중에
# game_type_list = []
assists_list = []
kills_list = []
deaths_list = []
champExp_list = [] #champ_experience_list
champName_list = [] #champion_name_list
indiPos_list = [] #individual_position_list
lane_list = []
role_list = []
gold_list = []   #gold_earned_list
teamPos_list = []
tDDealt_list = []  #total_damage_dealt_list
tDDealtChamp_list = [] #total_damage_dealt_to_champions_list
tDTaken_list = []  #total_damage_taken_list
tMKilled_list = []  #total_minions_killed_list
turKills_list = []  #turret_kills_list
win_list = []

for i in range(0,30):
    pair = list(match_id_dict.items())[i]
    puuid = pair[0]
    match_id = pair[1]
    url = f'https://asia.api.riotgames.com/lol/match/v5/matches/{match_id}?api_key={api_key}'

    # Make the Riot API call
    response = requests.get(url)
    response_json = response.json()

    assists = None  # set a default value for assists
    for participant in response_json['info']['participants']:
        if participant['puuid'] == puuid:
          #if participant['individualPosition'] == 'MIDDLE':
            assists = participant['assists']
            kills = participant['kills']
            deaths = participant["deaths"]
            champExp = participant["champExperience"]
            champName = participant["championName"]
            indiPos = participant["individualPosition"]
            teamPos = participant["teamPosition"]
            lane = participant["lane"]
            role = participant["role"]
            gold = participant["goldEarned"]
            tDDealt = participant["totalDamageDealt"]
            tDDealtChamp = participant["totalDamageDealtToChampions"]
            tDTaken = participant["totalDamageTaken"]
            tMKilled = participant["totalMinionsKilled"]
            turKills = participant["turretKills"]
            win = participant["win"]
            break

    if assists is not None:
        assists_list.append(assists)  # append assists only if it was set to a non-default value
    if kills is not None:
        kills_list.append(kills)
    if deaths is not None:
        deaths_list.append(deaths)
    if champExp is not None:
        champExp_list.append(champExp)
    if champName is not None:
        champName_list.append(champName)
    if indiPos is not None:
        indiPos_list.append(indiPos)
    if teamPos is not None:
        teamPos_list.append(teamPos)
    if lane is not None:
        lane_list.append(lane)
    if role is not None:
        role_list.append(role)
    if gold is not None:
        gold_list.append(gold)
    if tDDealt is not None:
        tDDealt_list.append(tDDealt)
    if tDDealtChamp is not None:
        tDDealtChamp_list.append(tDDealtChamp)
    if tDTaken is not None:
        tDTaken_list.append(tDTaken)
    if tMKilled is not None:
        tMKilled_list.append(tMKilled)
    if turKills is not None:
        turKills_list.append(turKills)
    if win is not None:
        win_list.append(win)
  
        




In [ ]:
mini_df = pd.DataFrame()
mini_df['assists'] = assists_list
mini_df['kills'] = kills_list
mini_df['deaths'] = deaths_list
mini_df['champExp'] = champExp_list #champ_experience_list
mini_df['champName'] = champName_list #champion_name_list
mini_df['indiPos'] = indiPos_list #individual_position_list
mini_df['lane'] = lane_list
mini_df['role'] = role_list 
mini_df['gold'] = gold_list    #gold_earned_list
mini_df['teamPos'] = teamPos_list 
mini_df['tDDealt'] = tDDealt_list  #total_damage_dealt_list
mini_df['tDDealtChamp'] = tDDealtChamp_list #total_damage_dealt_to_champions_list
mini_df['tDTaken'] = tDTaken_list  #total_damage_taken_list
mini_df['tMKilled'] = tMKilled_list   #total_minions_killed_list
mini_df['turKills'] = turKills_list   #turret_kills_list
mini_df['win'] = win_list 

In [ ]:
mini_df

assists  kills  deaths  champExp     champName  indiPos    lane     role  \
0         3      2       2     10214        Ezreal   BOTTOM  BOTTOM    CARRY   
1         1      4       3     12636          Sett      TOP  MIDDLE      DUO   
2         2      1      18      7679         Jayce      TOP     TOP     SOLO   
3         2      2       6      8414      MasterYi      TOP     TOP     SOLO   
4        15      8       2     10494        Xerath  UTILITY  BOTTOM  SUPPORT   
5         7      7       5     16986         Shaco      TOP     TOP     SOLO   
6         7      3       6      7495           Zac   JUNGLE  JUNGLE     NONE   
7         3     10       6     15900        KSante   MIDDLE  MIDDLE     SOLO   
8        19      7       8     13894  Heimerdinger  UTILITY  BOTTOM  SUPPORT   
9        15     16      22     25363          Sett  Invalid     TOP  SUPPORT   
10        9      3       7     12731         Xayah   BOTTOM  BOTTOM    CARRY   
11       22     18      11     21193       Kalista  Invalid  MIDDLE  SUPPORT   
12        3      7       8      8858        RekSai   JUNGLE  JUNGLE     NONE   
13        2      3       6     11252          Kled      TOP     TOP     SOLO   
14        5      7       5     14550           Zed   MIDDLE  MIDDLE     SOLO   
15       10      3      11     10075     Lissandra  UTILITY  BOTTOM  SUPPORT   
16        5      9       7     12890        LeeSin   JUNGLE  JUNGLE     NONE   
17       25      3       8     12627         Corki  Invalid    NONE  SUPPORT   
18        1      3       5     10955         Fiora      TOP     TOP     SOLO   
19        6     23       4     16938          Sett      TOP     TOP     SOLO   
20        4      3       8     11033          Nunu   MIDDLE  MIDDLE     SOLO   
21        3      5       6      6777          Ashe   BOTTOM  BOTTOM    CARRY   
22       34     17      10     20817        Veigar  Invalid    NONE  SUPPORT   
23       33      6      12     17293      JarvanIV  Invalid    NONE  SUPPORT   
24       28      8      10     19424    Blitzcrank  UTILITY  BOTTOM  SUPPORT   
25       10      9       7     15466         Riven      TOP     TOP     SOLO   
26        7     10       4     14606         Varus   BOTTOM  BOTTOM     SOLO   
27        2      7       5      8117        Ezreal   BOTTOM  BOTTOM    CARRY   
28       16      3       7     28468         Kayle      TOP     TOP      DUO   
29       12      0       7     15272        Ezreal   BOTTOM  BOTTOM    CARRY   

     gold  teamPos  tDDealt  tDDealtChamp  tDTaken  tMKilled  turKills    win  
0    8854   BOTTOM   112695         17186     9713       171         0  False  
1    8751      TOP    82084          9821    18498       167         0  False  
2    5819      TOP    64903         14541    30047        93         0  False  
3    6635      TOP    54874          6570    13566       105         0  False  
4   11108  UTILITY    90033         34020     9557        38         2   True  
5   12470      TOP   102270         22681    21841       139         4   True  
6    6473   JUNGLE    84530          8842    23139        23         0  False  
7   13053   MIDDLE   130734         35229    42056       175         1  False  
8   12559  UTILITY    70360         22903    24899        48         1   True  
9   16738            124726         49993    98979        70         0  False  
10  11561   BOTTOM   159724         17345    23801       198         1  False  
11  17100             93998         30965    33911        87         0   True  
12   9781   JUNGLE   126132         15519    24946        14         0  False  
13   8129      TOP    84376         12146    33041       138         0  False  
14  12911   MIDDLE   132620         21395    24370       170         0  False  
15   9497  UTILITY    58686         17208    22980        45         0  False  
16  11051   JUNGLE   187482         16545    34170        29         1  False  
17   9796             32219         14330    13648        23         0   True  
18 

# 앞으로 해야하는 것: 
##### 